source/extensions/access_loggers/grpc/http_grpc_access_log_impl.cc

```diff
@@ -120,7 +120,9 @@ void HttpGrpcAccessLog::emitLog(const Http::RequestHeaderMap& request_headers,
       if (!entry.empty()) {
         // TODO(https://github.com/envoyproxy/envoy/issues/13454): Potentially log all header
         // values.
-        logged_headers->insert({header.get(), std::string(entry[0]->value().getStringView())});
+        for (size_t i = 0; i < entry.size(); i++) {
+          logged_headers->insert({header.get(), std::string(entry[i]->value().getStringView())});
+        }
```

Comment:
Assuming that logged_headers is just a normal map, it still seems that only the first header will be successfully inserted. I think this may not achieve our purpose.

Reply:
if this is map i think then we just have to iterate map values and insert it ?!

Reply:
What I mean is that inserting multiple header values into the map with the same key does not get the result we expect: save all header values. 🤔

Reply:
ohh yeah got it. what should be the key values then ?!

Reply:
I also don't have any good ideas, this problem is not easy.  🤔

Reply:
ohh!! 
 @lizan  do you have idea for this mapped values logging ?!

Reply:
@ME-ON1 May be `HeaderUtility::getAllOfHeaderAsString` is a possible solution.

Reply:
for inserting the values in logged_headers map ?

Reply:
Yep. But But it is best to develop after the community accepts such a program.

Reply:
whom should i ask for GO !?

Reply:
Please add a test so it is much easier to see what's expected here for behaviors. `getAllOfHeaderAsString` could work if we decided in that way and documented correctly.